File: Source/Geometry/farthest_point.c
#include <math.h>

typedef struct {
    double x;
    double y;
} Point;

double calculateDistance(Point a, Point b) {
    return sqrt(pow(b.x - a.x, 2) + pow(b.y - a.y, 2));
}

Point findFarthestPoint(Point givenPoint, Point points[], int size) {
    double maxDistance = 0;
    Point farthestPoint;
    
    for(int i = 0; i < size; i++) {
        double distance = calculateDistance(givenPoint, points[i]);
        if(distance > maxDistance) {
            maxDistance = distance;
            farthestPoint = points[i];
        }
    }
    
    return farthestPoint;
}

File: Source/Geometry/farthest_point.js
class Point {
    constructor(x, y) {
        this.x = x;
        this.y = y;
    }
}

function calculateDistance(a, b) {
    return Math.sqrt(Math.pow(b.x - a.x, 2) + Math.pow(b.y - a.y, 2));
}

function findFarthestPoint(givenPoint, points) {
    let maxDistance = 0;
    let farthestPoint;
    
    for(let i = 0; i < points.length; i++) {
        let distance = calculateDistance(givenPoint, points[i]);
        if(distance > maxDistance) {
            maxDistance = distance;
            farthestPoint = points[i];
        }
    }
    
    return farthestPoint;
}
